e2e/_suites/kubernetes-autodiscover/features/filebeat.feature

```diff
@@ -1,38 +1,35 @@
-@kubernetes
-@autodiscover
+@kubernetes-autodiscover
 @filebeat
-Feature: Log collection from Kubernetes pods.
+Feature: Filebeat
   Use Kubernetes autodiscover features in Filebeat to collect logs.
 
-Scenario: Logs collection from running pod
+Background:
   Given "filebeat" is running with "hints enabled"
+
+Scenario: Logs collection from running pod
    When "a pod" is deployed
    Then "filebeat" collects events with "kubernetes.pod.name:a-pod"
 
 Scenario: Logs collection from failing pod
-  Given "filebeat" is running with "hints enabled"
    When "a failing pod" is deployed
    Then "filebeat" collects events with "kubernetes.pod.name:a-failing-pod"
 
 # This scenario explicitly waits for 60 seconds before doing checks
 # to be sure that at least one job has been executed.
 Scenario: Logs collection from short-living cronjobs
-  Given "filebeat" is running with "hints enabled"
-   When "a short-living cronjob" is deployed
-    And "60s" have passed
+  Given "a short-living cronjob" is deployed
+   When "60s" have passed
```

Comment:
I'd keep it like this, because the `When` clause is considered the real test, being the `Given` the requirements/preparation for the scenario. The `Given` is optional, `When`, mandatory.

```suggestion
  When "a short-living cronjob" is deployed
   And "60s" have passed
```

Reply:
I agree with this, this is what I had it before, but `gherkin-lint` doesn't think the same:
```
AvoidScripting - Multiple Actions
  e2e/_suites/kubernetes-autodiscover/features/filebeat.feature (19): Filebeat.Logs collection from short-living cronjobs
```

Reply:
Let's go back to what the linter wants, I don't see it so bad and this way we don't need to add an exception.